src/types.mligo

```diff
@@ -51,8 +51,16 @@ type nonce = nat
 // Represents whether a voter has voted against (false) or for (true) a given proposal.
 type vote_type = bool
 
-// Stores all voter data for a proposal
-type voter_map = (address * vote_type, nat) map
+type vote_item =
+  { vote_amount : nat
+  ; vote_type : vote_type
+  }
+
+type vote_status =
+  { votes : vote_item list
```

Comment:
I see that we have replced `voter_map` with `vote_item list`, I suppose this is done to track the total number of staked tokens. If that is the case can't we just add a `nat` field to `vote_status` to keep track of staked tokens, and increment it during each `vote` operation for this user, and keep the `votes` field as `voter_map` itself?

Reply:
`type voter_map = (address * vote_type, nat) map` is replaced by `(address * proposal_key, vote_status) big_map`, and is moved outside of the `proposal` to `storage`.

`vote_status` type is as followed:
```
type vote_status =
  { votes : vote_item list
  ; is_unfreezed : bool
  }

type vote_item =
  { vote_amount : nat
  ; vote_type : vote_type
  }

 ```
 
 I'm not sure exactly what your suggestion is, but I guess you mean that it is better to just use `nat` instead of a `list` there.
 
 ```
 type vote_status =
  { upvotes : nat
  ; downvotes: nat
  ; is_unfreezed : bool
  }
```

I could do it this way as well, but I think storing each upvote and downvote as a list could be useful in the future but I may be wrong.

Reply:
```
type vote_status =
 { upvotes : nat
 ; downvotes: nat
 ; is_unfreezed : bool
 }
```
Ah, yes, that would work better.

>I think storing each upvote and downvote as a list could be useful in the future but I may be wrong.

But we didn't have that capability as of now, right?

Reply:
> But we didn't have that capability as of now, right?
The current implementation already stores upvote and downvote as a list already. You mean we don't have the usage of it now right? 

@pasqu4le Since you suggest keeping the vote history, what do you think about @sras feedback? With the new feedback, we still keep track of the vote history but we won't know how many times the voter upvote/downvote a proposal (which may or may not be useful).

Reply:
>The current implementation already stores upvote and downvote as a list already.

@rinn7e Oh, by current implementation, I meant the version in master branch. AFAIU I that version, we don't keep the votes in a list.

Reply:
I mean the implementation in this PR already store them in a list.

This is due to the suggestion 3 in here https://github.com/tezos-commons/baseDAO/pull/313#discussion_r767946169

Reply:
Oh. I didn't see that comment. Hope he can provide some context.

Reply:
> Hope he can provide some context

I mentioned this elsewhere as well, but this was a requested feature: #151

Reply:
@pasqu4le the issue is that version in master branch stores the information in  a map, which in this PR it is changed to a list. So basically the difference is that earlier we indeed stored who voted up or down, but didn't store how many times they voted. That is if they voted two times with 5 upvotes, then we only stored that they voted with 10 upvotes, but not that it was 2 times, with 5 votes each. But this PR adds this additional information, because the votes are stored in a list, so each instance of a vote is stored, which is more information than we need.

This is the change that I am pointing out as being new.

Reply:
Regarding this point, I'm good with both combining the votes (like the previous implementation) or keeping it as a list (this PR implementation). I use the list simply because it could be useful, but if it isn't and cost too much, I'll go with combining them instead.

Reply:
Ah I see, I didn't understand what was the difference, sorry.

AFAIK we only _need_ to keep the total amount of upvotes/downvotes and not the historic list of votes submitted.

I agree that we shouldn't keep this as we have no use for it now, so it looks like indeed:
```
type vote_status =
 { upvotes : nat
 ; downvotes: nat
 ; is_unfreezed : bool
 }
```
would be more efficient and work out just fine for this.

Reply:
I've updated this. I think this is ready for re-review.